src/ucp/core/ucp_ep.c

```diff
@@ -2183,7 +2183,7 @@ ucp_ep_config_set_am_rndv_thresh(ucp_worker_h worker,
     ucs_assert(config->key.am_lane != UCP_NULL_LANE);
     ucs_assert(config->key.lanes[config->key.am_lane].rsc_index != UCP_NULL_RESOURCE);
 
-    if (context->config.ext.rndv_thresh == UCS_MEMUNITS_AUTO) {
+    if (context->config.ext.rndv_inter_thresh == UCS_MEMUNITS_AUTO) {
```

Comment:
Yes, that's the idea.

Basically I considered 2 approaches:
1. Keep common variable `rndv_thresh` together with fine-grained settings.
(+) This way we could keep the existing code the same
(-) It would complicate the key-value parser by adding a concept of "common config" offset

2. Replace common config with fine-grained settings
(-) We need to adapt the existing code for protov1
(+) We have cleaner design for key-value feature

I raised this question last year during the meeting, and IIRC Yossi supported the second approach, therefore this change

Reply:
Completely agree with the current approach, but I have an idea how to possibly improve it.

I suggest to add `proto_disable_rndv_thresh` field to `ucp_context_config` structure, set it to `rndv_inter_thresh` inside `ucp_fill_config()`, also leave the comment with something like "this variable is used for protov1 only because protov2 supports different thresholds for inter/intra nodes" near the initialization place and then use this variable in all protov1 related places.

I understand that this is debatable suggestion because of extra-field which is used only for better code self-documentation so it is up to you.

But anyway I think it makes sense to insert some comments about that and maybe check somewhere whether we use protov1 and have different inter/intra thresholds and print warning in that case.